### ISSUE TITLE
10$: disallow (a&&b??c)

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#933]
`&&` does not associate with `??`. We should report a nice diagnostic for code like the following:
```javascript
let a;
let b;
let c;
a && b ?? c;
```
